plugin/src/main/java/com/playmonumenta/scriptedquests/zones/zonetree/ParentZoneTree.java

```diff
@@ -0,0 +1,170 @@
+		 */
+
+		 class ParentData {
+			public int mPriority;
+			public Axis mAxis;
+			public double mPivot;
+			public double mMidMin;
+			public double mMidMax;
+			public ArrayList<ZoneFragment> mLess = new ArrayList<ZoneFragment>();
+			public ArrayList<ZoneFragment> mMid = new ArrayList<ZoneFragment>();
+			public ArrayList<ZoneFragment> mMore = new ArrayList<ZoneFragment>();
+		}
+
+		// Default is an impossibly worst case scenario so it will never be chosen.
+		ParentData bestSplit = new ParentData();
+		bestSplit.mPriority = zones.size();
+
+		for (ZoneFragment pivotZone : zones) {
+			for (Axis axis : AXIS_ORDER) {
+				double[] possiblePivots = new double[2];
+				possiblePivots[0] = ZoneUtils.vectorAxis(pivotZone.minCorner(), axis);
+				possiblePivots[1] = ZoneUtils.vectorAxis(pivotZone.trueMaxCorner(), axis);
+				for (double pivot : possiblePivots) {
+					ParentData testSplit = new ParentData();
+					testSplit.mAxis = axis;
+					testSplit.mPivot = pivot;
+					testSplit.mMidMin = pivot;
+					testSplit.mMidMax = pivot;
+
+					for (ZoneFragment zone : zones) {
+						if (pivot >= ZoneUtils.vectorAxis(zone.trueMaxCorner(), axis)) {
+							testSplit.mLess.add(zone);
+						} else if (pivot >= ZoneUtils.vectorAxis(zone.minCorner(), axis)) {
+							testSplit.mMidMin = Math.min(testSplit.mMidMin,
+							                             ZoneUtils.vectorAxis(zone.minCorner(), axis));
+							testSplit.mMidMax = Math.max(testSplit.mMidMax,
+							                             ZoneUtils.vectorAxis(zone.trueMaxCorner(), axis));
+							testSplit.mMid.add(zone);
+						} else {
+							testSplit.mMore.add(zone);
+						}
+					}
+
+					testSplit.mPriority = Math.max(testSplit.mLess.size(), testSplit.mMore.size());
+					testSplit.mPriority = Math.max(testSplit.mPriority, testSplit.mMid.size());
+
+					if (testSplit.mPriority < bestSplit.mPriority) {
+						bestSplit = testSplit;
+					}
+				}
+			}
+		}
+
+		// This is the answer we want. Copy values to self.
+		mAxis = bestSplit.mAxis;
+		mPivot = bestSplit.mPivot;
+		mMidMin = bestSplit.mMidMin;
+		mMidMax = bestSplit.mMidMax;
+
+		if (bestSplit.mPriority >= zones.size()) {
+			/*
+			 * The priority of our best case scenario is equal to or worse than our worst case.
+			 *
+			 * This should only occur if two zone fragments overlap, which is prevented by earlier
+			 * sections of code. If this occurs, then somehow zone fragments are not being divided
+			 * or handled properly.
+			 */
+			sendError(sender, "A serious plugin error has occured. Zone fragments dropped to keep us going.");
+			sendError(sender, "Zones:");
+			for (ZoneFragment zone : zones) {
+				sendError(sender, "- " + zone.toString());
+			}
+			ArrayList<ZoneFragment> tempDebugList = new ArrayList<ZoneFragment>();
+			tempDebugList.add(zones.get(0));
+			mLess = CreateZoneTree(sender, tempDebugList);
+			mMid = CreateZoneTree(sender, tempDebugList);
+			mMore = CreateZoneTree(sender, tempDebugList);
```

Comment:
Do we really want to recover from this error? Maybe it should just fail? Up to you, just an idea... failing makes it more likely to be reported and fixed quickly.

Reply:
I can throw an error all the way up, which I think I did in an earlier commit. At some point I got horrified by how many places I wound up catching and handling exceptions, though.

Reply:
I wasn't thinking about error handling, actually - just... not continuing to recurse with the tempDebugList. Catch the error here, print it to the sender and the logger, and... just stop instead of trying to continue excluding the failing zone lists.